RDF triple based deletion

ArkanEmre opened the pull request.

Commit:
add DirectStatementDeleter for deletion
add DeleteRDF and DeleteResults to RDFImport
add test for deletion to RDFImportTest
add test datasets for deletion

Commit:
edit comparision to be handeled by RDF4J

Commit:
add new tests for deletion
add info for not deleted statements e.g. blank nodes
rename test files for consistency
add new test files

rnavarropiris commented on src/main/java/semantics/DirectStatementDeleter.java:
```diff
@@ -0,0 +1,293 @@
+import org.neo4j.logging.Log;
+
+/**
+ * Created by jbarrasa on 09/11/2016.
```
Please fix this to reflect the real authorship and creation date

rnavarropiris commented on src/main/java/semantics/DirectStatementDeleter.java:
```diff
@@ -0,0 +1,293 @@
+      Iterable it = (Iterable) value;
+      Object first = Iterables.firstOrNull(it);
+      if (first == null) {
+        return EMPTY_ARRAY;
```
Since the returned array is modifiable, posterior calls might be compromised.
Please return `new String[0]` and remove the static variable

rnavarropiris commented on src/main/java/semantics/RDFImport.java:
```diff
@@ -336,6 +336,67 @@ private int getHandleMultivalAsInt(String ignoreUrlsAsText) {
+      RDFParser rdfParser = Rio.createParser(getFormat(format));
+      rdfParser.setRDFHandler(statementDeleter);
+      rdfParser.parse(inputStream, url);
+    } catch (MalformedURLException e) {
```
Why is this exception not part of the next catch?
Shouldn't the status be KO as well in this case?

Commit:
fix unit test, minor class format improvements, defensive return of empty array

Commit:
add inline comment regarding recognized error

Commit:
change DT of notDeletedStatementCount to long
add counter for blank nodes
add class description
fix number of deleted triples

Commit:
Merge branch 'develop' into RDFDelete
